aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -134,7 +134,184 @@ For more information, see <xref:blazor/index>.
+
+```csharp
+[Authorize]
+public class ChatHub : Hub
+{
+    public void SendMessage(string message)
+    {
+    }
+
+    [Authorize("DomainRestricted")]
+    public void BanUser(string username)
+    {
+    }
+
+    [Authorize("DomainRestricted")]
+    public void ViewUserHistory(string username)
+    {
+    }
+}
+```
+
+Creating the `DomainRestricted` policy is as simple as wiring it up using the authorization middleware. In `Startup.cs`, add the new policy, providing the custom `DomainRestrictedRequirement` requirement as a parameter.
+
+```csharp
+services
+    .AddAuthorization(options =>
+    {
+        options.AddPolicy("DomainRestricted", policy =>
+        {
+            policy.Requirements.Add(new DomainRestrictedRequirement());
+        });
+    });
+```
+
+We are hooking SignalR hubs into the new [Endpoint Routing](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing) feature recently released. SignalR hub wire-up was previously done explicitly:
+
+```csharp
+app.UseSignalR(routes =>
+{
+    routes.MapHub<ChatHub>("hubs/chat");
+});
+```
+
+This meant developers would need to wire up controllers, Razor pages, and hubs in a variety of different places during startup, leading to a series of nearly-identical routing segments:
+
+```csharp
+app.UseSignalR(routes =>
+{
+    routes.MapHub<ChatHub>("hubs/chat");
+});
+
+app.UseRouting(routes =>
+{
+    routes.MapRazorPages();
+});
+```
+
+Now, SignalR hubs can also be routed via endpoint routing, so you’ve got a one-stop place to route nearly everything in ASP.NET Core.
+
+```csharp
+app.UseRouting(routes =>
+{
+    routes.MapRazorPages();
+    routes.MapHub<ChatHub>("hubs/chat");
+});
+```
+
+With ASP.NET Core SignalR we added Streaming support, which enables streaming return values from server-side methods. This is useful for when fragments of data will come in over a period of time.
+
+With .NET Core 3.0 we’ve also added client-to-server streaming. With client-to-server streaming, your server-side methods can take instances of a `ChannelReader<T>`. In the C# code sample below, the `StartStream` method on the Hub will receive a stream of strings from the client.
+
+```csharp
+public async Task StartStream(string streamName, ChannelReader<string> streamContent)
+{
+    // read from and process stream items
+    while (await streamContent.WaitToReadAsync(Context.ConnectionAborted))
+    {
+        while (streamContent.TryRead(out var content))
+        {
+            // process content
+        }
+    }
+}
```

Comment:
I would use the new IAsyncEnumerable sample here, since it's simpler than the ChannelReader version:

https://github.com/aspnet/AspNetCore.Docs/blob/4aa859cb075f3b1b250aeccfcbcabd5e810b974e/aspnetcore/signalr/streaming.md#L88-L106

Reply:
@halter73  can you make that change in a commit on #14371 ?